File: Daily_Questions/Strings/409.cpp
// // https://leetcode.com/problems/longest-palindrome/description/

// class Solution
// {
//     public:
//         int longestPalindrome(string s)
//         {

//             int count = 0;
//             if (s.size() == 1)
//             {
//                 return 1;
//             }
//             else
//             {
//                 unordered_map<char, int> freq;
//                 for (char c: s)
//                 {
//                     freq[c]++;
//                 }
//                 for (auto it = freq.begin(); it != freq.end(); ++it)
//                 {
//                     cout << it->first << ": " << it->second << endl;
//                 }

//                 bool isOdd = false;
//                 for (auto it = freq.begin(); it != freq.end(); it++)
//                 {
//                     if (it->second % 2 == 0)
//                     {
//                         count = count + it->second;
//                     }
//                     else
//                     {
//                         count = count + it->second - 1;
//                         isOdd = true;
//                     }
//                 }

//                 if (isOdd)
//                 {
//                     count = count + 1;
//                 }

//                 return count;
//             }
//         }
// };

File: Daily_Questions/Strings/409.java
// https://leetcode.com/problems/longest-palindrome/


// class Solution {
//     public int longestPalindrome(String s) {
//         int count = 0;
//         if (s.length() == 1) {
//             return 1;
//         } else {
//             Map<Character, Integer> freq = new HashMap<>();
//             for (char c : s.toCharArray()) {
//                 freq.put(c, freq.getOrDefault(c, 0) + 1);
//             }

//             boolean hasOdd = false; // To track if there's at least one character with an odd count
//             for (Map.Entry<Character, Integer> entry : freq.entrySet()) {
//                 if (entry.getValue() % 2 == 0) {
//                     count += entry.getValue();
//                 } else {
//                     count += entry.getValue() - 1; // Add the even part of the count
//                     hasOdd = true;
//                 }
//             }

//             if (hasOdd) {
//                 count += 1; // Add one center character if there's any odd count character
//             }

//             return count;
//         }
//     }}
